### PR TITLE
New version of bowtie2

### PR DESCRIPTION
**Pull request by jwarnn**
- [ ] This comment contains a description of what is in the pull request.

- [ ] Build your own docker image using a Dockerfile
  - [ ] Directory structure should be name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. spades/3.12.0/Dockerfile)
  - [ ] Includes the recommended LABELS
- [ ] (Optional) Dockerfile is built with best practices and has been approved by a linter (such as https://hadolint.github.io/hadolint/)
- [ ] Edit main README.md
- [ ] Edit Program_Licenses.md
- [ ] Create a simple container-specific README.md in the same directory as the Dockerfile (i.e. spades/3.12.0/README.md)
- [ ] Write a GitHub actions workflow
  - [ ] Should be located in .github/workflows/ and named test-<tool>.yml (i.e. .github/workflows/test-spades.yml)
  - [ ] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
  - [ ] Have successfully run the workflow "Test <program name> image" in your forked repository

- [ X] Build your own docker image using a Dockerfile
  - [ X] Directory structure should be name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. spades/3.12.0/Dockerfile)
  - [ X] Includes the recommended LABELS
- [ ] (Optional) Dockerfile is built with best practices and has been approved by a linter (such as https://hadolint.github.io/hadolint/)
- [X ] Edit main README.md
- [ X] Ensure tool is listed in Program_Licenses.md
- [ X] Create a simple container-specific README.md in the same directory as the Dockerfile (i.e. spades/3.12.0/README.md)
- [ X] Update GitHub actions workflow if needed
- [ X] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
- [ X] Have successfully run the workflow "Test <program name> image" in your forked repository

- [ ] Build your own docker image using a Dockerfile
  - [ ] Includes the recommended LABELS
- [ ] (Optional) Dockerfile is built with best practices and has been approved by a linter (such as https://hadolint.github.io/hadolint/)
- [ ] Ensure tool is listed in Program_Licenses.md
- [ ] Ensure a simple container-specific README.md exists in the same directory as the Dockerfile (i.e. spades/3.12.0/README.md)
- [ ] Update GitHub actions workflow if needed
- [ ] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
- [ ] Have successfully run the workflow "Test <program name> image" in your forked repository

- [ ] Update relevant GitHub actions workflow files
- [ ] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
- [ ] Have successfully run the workflow "Test <program name> image" in your forked repository

I used the previous version of bowtie2 as a backbone for my dockerfile, thanks Holly Halstead holly.halstead@doh.wa.gov,  but updated version and added a test target as well.

### CODE REVIEW

**Comment by erinyoung** [comment]
Hmmm... For some reason the github actions didn't kick off. Let me look into this.

**Comment by kapsakcj** [comment]
Thanks for the PR! Appreciate you including the GH Actions workflows

> Hmmm... For some reason the github actions didn't kick off. Let me look into this.

Might be due to the formatting of the YML. Does not look properly spaced to me. This YML `.github/workflows/test-bowtie2.yml`

File: bowtie2/2.4.5/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+CMD ["bowtie2"]
+
+FROM app as test
+
+RUN mkdir /index
+RUN wget https://raw.githubusercontent.com/BenLangmead/bowtie2/master/example/reads/longreads.fq
+RUN wget https://raw.githubusercontent.com/BenLangmead/bowtie2/master/example/reference/lambda_virus.fa
+RUN bowtie2-build lambda_virus.fa /index/lambda_virus
+RUN bowtie2 -x /index/lambda_virus -U longreads.fq
+RUN bowtie2-inspect --summary /index/lambda_virus
```
Comment (kapsakcj) [inline bowtie2/2.4.5/Dockerfile:66]: I would recommend to combine all the RUN layers in lines 61-66 into one to consolidate into one. something like this:
```Dockerfile
RUN mkdir /index && \
 wget https://raw.githubusercontent.com/BenLangmead/bowtie2/master/example/reads/longreads.fq  && \
 wget https://raw.githubusercontent.com/BenLangmead/bowtie2/master/example/reference/lambda_virus.fa  && \
 bowtie2-build lambda_virus.fa /index/lambda_virus  && \
 bowtie2 -x /index/lambda_virus -U longreads.fq && \
 bowtie2-inspect --summary /index/lambda_virus
```

**Comment by kapsakcj** [comment]
Please rename `bowtie2/2.4.5/READ.me` to `bowtie2/2.4.5/README.md` so it is properly rendered in github

**Comment by jwarnn** [comment]
Thanks for the clear comments! I updated the READ.md and changed the file name, then I made the change to the docker file on how the commands were called for the test, and I fixed the indentation formatting for the test-bowtie2.yml. Anything else I need to do to kick off the workflows?

**Comment by fanninpm** [comment]
> READ.md

Because the file is **not** named `README.md`, [GitHub does not automatically display it when a user navigates to the directory it resides in.](https://github.com/StaPH-B/docker-builds/tree/6c97c3c49548c3429e7fe5d158fadabc3f6047f5/bowtie2/2.4.5)

Compare to [this directory, which has a `README.md`](https://github.com/StaPH-B/docker-builds/tree/master/parsnp).

**Pull request merged**